src/components/GettingStartedModal/GroupRegistrationStep.tsx

```diff
@@ -0,0 +1,100 @@
+import { ChangeEvent, FormEvent, useState } from "react";
+import { NavigationButtons, Step, StepProps } from ".";
+import { GROUPS } from "../../state/reducers/groupId";
+import { Button } from "../Button";
+
+export const GroupRegistrationStep: Step = {
+  title: "Personal Information",
+  Component: GroupRegistrationForm,
+  commitState: ({ groupId, email, whereFound }, { registerGroup, setUserEmail, setWhereFound }) => {
+    registerGroup(groupId);
+    setWhereFound(whereFound);
+    setUserEmail(email);
+  },
+};
+
+export function GroupRegistrationForm({
+  wizardState: { groupId , email, whereFound},
+  setWizardState,
+  goToNextStep,
+  goToPreviousStep,
+  exitWizard
+}: StepProps) {
+  const [enabled, setEnabled] = useState(groupId!=undefined && email !== ''); 
+  
+  function onRadioChanged(e: ChangeEvent<HTMLInputElement>) {
+    const groupId = e.target.value;
+    setWizardState((s) => ({ ...s, groupId }));
+    setEnabled(email !== ''); 
+  }
+
+  function onEmailChanged(e: ChangeEvent<HTMLInputElement>) {
+    const email = e.target.value;
+    setWizardState((s) => ({ ...s, email }));
+    setEnabled(groupId !== undefined && email !== ''); 
+  }
+
+  function onWhereFoundChanged(e: ChangeEvent<HTMLInputElement>){
+    const whereFound = e.target.value;
+    setWizardState((s) => ({ ...s, whereFound }));
+    setEnabled(whereFound !== undefined);
+  }
+
+  function onSubmit(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    if (groupId && email && goToNextStep) goToNextStep();
+  }
+  return (
+    <>
+      <p> 
+        We're happy to have you here! Who are you, and how did you hear about us?
+        <br></br>
+        <br></br>
+        After you've completed this, you're all set! Continue for advanced settings.
+      </p>
+      <form onSubmit={onSubmit}>
+        <fieldset>
+          <legend>Email Address</legend>
+          <input
+            type = "email"
+            name = "email"
+            value = {email}
+            onChange = {onEmailChanged}
+          />
+        </fieldset>
+        <fieldset>
+          <legend>Where did you find us?</legend>
+          <input
+            type = "whereFound"
+            name = "whereFound"
+            value = {whereFound}
+            onChange = {onWhereFoundChanged}
+          />
+        </fieldset>
```

Comment:
So this is a weird HTML moment, but I don't think it's semantically correct to use `fieldset` and `legend` with a single input. The input is supposed to have a `label` that explains what it is, not a legend.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset

Maybe these two text inputs should be in a fieldset together and then each have their own label? That or they should be loose in the form or in `div`s.

Reply:
This has been changed to make sense within the context of a fieldset.